Mathlib/Order/Interval/Set/Infinite.lean

```diff
@@ -16,15 +16,13 @@ preorder is an infinite type.
 
 variable {α : Type*} [Preorder α]
 
-/-- A nonempty preorder with no maximal element is infinite. This is not an instance to avoid
-a cycle with `Infinite α → Nontrivial α → Nonempty α`. -/
-theorem NoMaxOrder.infinite [Nonempty α] [NoMaxOrder α] : Infinite α :=
+/-- A nonempty preorder with no maximal element is infinite. -/
+instance NoMaxOrder.infinite [Nonempty α] [NoMaxOrder α] : Infinite α :=
```

Comment:
What happens to the benchmark if we do:
```suggestion
/-- A nonempty preorder with no maximal element is infinite.
This instance has lower priority to avoid the cycle `Infinite α → Nontrivial α → Nonempty α` if possible.
-/
instance (priority := 100) NoMaxOrder.infinite [Nonempty α] [NoMaxOrder α] : Infinite α :=
```